hibernate-spatial/src/main/java/org/hibernate/spatial/dialect/postgis/PostgisPG9Dialect.java

```diff
@@ -48,6 +48,12 @@ public void contributeTypes(TypeContributions typeContributions, ServiceRegistry
 		support.contributeTypes( typeContributions, serviceRegistry );
 	}
 
+	@Override
+	public boolean equivalentTypes(int typeCode1, int typeCode2) {
+		return super.equivalentTypes( typeCode1, typeCode2 ) ||
+				( support.isSpatial( typeCode1 ) && support.isSpatial( typeCode2 ) );
+	}
+
```

Comment:
Could we move the logic into the parent class to avoid duplication in each child dialect class?

Reply:
Saw it relies on 'support', which is not put into parent class. Ignore my above comment if necessary.